GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -0,0 +1,368 @@
+                        this.localUpgradeServices.RunInstaller(installerPath, entry.Args, out installerExitCode, out localError);
+
+                        installSuccesesfull = installerExitCode == 0;
+
+                        // Just for initial experiment to make sure each step is displayed in UI
+                        Thread.Sleep(5 * 1000);
+
+                        return installSuccesesfull;
+                    },
+                    $"Installing {entry.Name} Version: {entry.Version}");
+            }
+
+            if (!installSuccesesfull)
+            {
+                error = localError;
+                return false;
+            }
+            else
+            {
+                error = null;
+                return true;
+            }
+        }
+
+        public bool TrySetupToolsDirectory(out string upgraderToolPath, out string error)
+        {
+            return this.localUpgradeServices.TrySetupToolsDirectory(out upgraderToolPath, out error);
+        }
+
+        private static bool TryGetPersonalAccessToken(string gitBinaryPath, string credentialUrl, ITracer tracer, out string token, out string error)
+        {
+            GitProcess gitProcess = new GitProcess(gitBinaryPath, null, null);
+            return gitProcess.TryGetCredentials(tracer, credentialUrl, out string username, out token, out error);
+        }
+
+        private string UnzipPackageToTempLocation()
+        {
+            string extractedPackagePath = this.localUpgradeServices.TempPath;
+            ZipFile.ExtractToDirectory(this.downloadedPackagePath, extractedPackagePath);
+            return extractedPackagePath;
+        }
+
+        public class NugetUpgraderConfig
+        {
+            public NugetUpgraderConfig(ITracer tracer, LocalGVFSConfig localGVFSConfig)
+            {
+                this.Tracer = tracer;
+                this.LocalConfig = localGVFSConfig;
+            }
+
+            public NugetUpgraderConfig(
+                ITracer tracer,
+                LocalGVFSConfig localGVFSConfig,
+                string feedUrl,
+                string packageFeedName,
+                string feedUrlForCredentials)
+                : this(tracer, localGVFSConfig)
+            {
+                this.FeedUrl = feedUrl;
+                this.PackageFeedName = packageFeedName;
+                this.FeedUrlForCredentials = feedUrlForCredentials;
+            }
+
+            public string FeedUrl { get; private set; }
+            public string PackageFeedName { get; private set; }
+            public string FeedUrlForCredentials { get; private set; }
+            private ITracer Tracer { get; set; }
+            private LocalGVFSConfig LocalConfig { get; set; }
+
+            public bool TryLoad(out bool isEnabled, out bool isConfigured, out string error)
+            {
+                error = string.Empty;
+                isEnabled = false;

```

Comment:
It's ok that you are setting `error` here, as you are appending to it later. However, `isEnabled` and `isConfigured` are set unconditionally further down, so do not need to be set here.